besu/src/main/java/org/hyperledger/besu/cli/options/stable/DataStorageOptions.java

```diff
@@ -52,6 +55,25 @@ public class DataStorageOptions implements CLIOptions<DataStorageConfiguration>
       arity = "1")
   private final Long bonsaiMaxLayersToLoad = DEFAULT_BONSAI_MAX_LAYERS_TO_LOAD;
 
+  @CommandLine.ArgGroup(validate = false)
+  private final DataStorageOptions.Unstable unstableOptions = new Unstable();
+
+  static class Unstable {
+    @CommandLine.Option(
+        hidden = true,
+        names = {"--Xbonsai-trie-log-retention-threshold"},
+        description =
+            "The number of blocks for which to retain trie logs. 0 is unlimited. (default: ${DEFAULT-VALUE})")
+    private long bonsaiTrieLogRententionThreshold = DEFAULT_BONSAI_TRIE_LOG_RETENTION_THRESHOLD;
+
+    @CommandLine.Option(
+        hidden = true,
+        names = {"--Xbonsai-trie-log-prune-limit"},
```

Comment:
I understood that we don't want to use pruning ? And for me the name and the description doesn't match a lot maybe `retention_trie_log_cache_limit`?

Reply:
not sure about my proposition

Reply:
This is not the same as the retention threshold so would be concerned about confusion there. Perhaps we need a better overall feature name to group them.
 
This option is being used both as a cache loading limit as well as a prune limit, with the loading limit being the more significant factor in performance. I like "cache-limit", although not strictly true technically the cache can grow higher than this limit, especially with the finalization changes.

Reply:
So we have:
--Xbonsai-trie-log-pruning-enabled
and
--Xbonsai-trie-log-pruning-limit

where "bonsai-trie-log-pruning" is like the feature name.

This pruning limit is being used as the "cache preload limit" (`loadingLimit`) as well as the "pruning window limit": https://github.com/siladu/besu/blob/3a779ea53ed2c9a2b438eb91c2d85231ebd59b1d/ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/trielog/TrieLogPruner.java#L133

I think `bonsai-trie-log-cache-preload-limit` could equally work and still be accurate.

`bonsai-trie-log-cache-limit` is perhaps a little misleading.

Reply:
is this cache is used also for block processing or validation bonsai-trie-log-cache-preload-limit ?

Reply:
just trying to understand if it's only used for the pruning or also for trielog access in general

Reply:
> is this cache is used also for block processing or validation bonsai-trie-log-cache-preload-limit ?
This cache is only accessed by the trieLogPruner and trieLogManager, which adds to it upon saving a trie log.

> just trying to understand if it's only used for the pruning or also for trielog access in general
yes, only pruning